src/app/dataset-view/additional-components/metadata-component/components/form-components/schema-field/schema-field.component.ts

```diff
@@ -15,11 +38,92 @@ export interface SchemaType {
     styleUrls: ["./schema-field.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SchemaFieldComponent extends BaseField {
-    public displayedColumns: string[] = ["name", "type", "actions"];
-    public availableTypes = ["STRING", "BIGINT", "TIMESTAMP"];
-    private defaultType = "STRING";
+export class SchemaFieldComponent extends BaseField implements AfterViewInit {
+    private defaultType = "";
     @ViewChild(MatTable) table: MatTable<unknown>;
+    @ViewChildren("ngbTypeahed") ngbType: QueryList<NgbTypeahead>;
+    public focus$ = new Subject<SchemaType>();
+    public click$ = new Subject<SchemaType>();
+    private focusObservableList = Array<Subject<SchemaType>>();
+    private clickObservableList = Array<Subject<SchemaType>>();
+    private ngbTypeHeader: NgbTypeahead;
+
+    public displayedColumns: string[] = ["name", "type", "actions"];
+    public availableTypes = [
+        "STRING",
+        "BIGINT",
+        "TIMESTAMP(p)",
+        "BOOLEAN",
+        "INT",
+        "DECIMAL(p,s)",
+        "FLOAT",
+        "DOUBLE",
+        "UUID",
+        "DATE",
+        "TIME",
+    ];
+
+    ngAfterViewInit(): void {
+        let focusItem$ = new Subject<SchemaType>();
+        this.trackSubscription(
+            this.focus$.subscribe((item) => {
+                const tableIndex = this.items.controls.findIndex(
+                    (element: AbstractControl) =>
+                        element.value.name === item.name,
+                );
+                this.focusObservableList.some((focus, index) => {
+                    if (index === tableIndex) {
+                        focusItem$ = focus;
+                        return true;
+                    } else {
+                        return false;
+                    }
+                });
+                focusItem$.next(item);
+            }),
+        );
+    }
+
+    public search: OperatorFunction<string, readonly string[]> = (
```

Comment:
Please decompose this function into a few smaller ones, it is too difficult to read. I assume we can split by subject.

Reply:
Ok